protocol/protocol.proto

```diff
@@ -497,6 +497,7 @@ message MediaPayload {
         int32 chunk_count = 3;
         StreamSettings settings = 4;
         optional bytes space_id = 5;
+        optional PublicScope public_scope = 6;
```

Comment:
Can you explain what this scope is for? Currently it serves no purpose.

Reply:
yes - scope is used to signal that this is a public media channel for storing content. Anyone can decrypt the content stored in this stream with a key that will be openly available. Like a Town's image. This unblocks scenario like NFT image. 
There will be 2 such publicly scoped media streams. One for Town; the other for User's avatar image.

Reply:
Signal to what or to whom? The fact that keys are or are not available to publicly download has no bearing on the internal state of this structure. 

Asked in another way - how does code change if you delete this property? (Aside from the checks, but changing that is fine)

Reply:
PublicScope is part of the protocol because the normal group encryption algo does not apply in a public stream. The public media stream will be using a symmetric key for encryption. Only the space owner is allowed to upload to the stream. Anyone can use the public symmetric key to decrypt content in the stream.

So besides the checks, encryption / decryption message processing is also different at the protocol level. Haven't work out the rest of the details yet.

Here is the next chunk of changes with a test that shows how the stream owner will encrypt and upload to the public stream: https://github.com/river-build/river/pull/395